### PR TITLE
Feature/offline download missed out features - [iOS]

### PR DESCRIPTION
**Pull request by PratheeshDBennet**
Hi Brian, 

Please find the PR details, 

Added the below functionalities to the RN wrapper. These logics are added to the bridging file BrightcovePlayerUtil where the legacy download logics were already there. 

- Pause a download
- Resume a download
- Cancel a download

Made the changes in the example App.js as well to demonstrate the above newly added features. 

Thanks

### CODE REVIEW

File: ios/BrightcovePlayerUtil.m [hunk 1]
```diff
@@ -99,10 +105,43 @@ - (void)stopObserving {
...
+RCT_EXPORT_METHOD(pauseVideoDownload:(NSString *)accountId policyKey:(NSString *)policyKey videoToken:(NSString *)videoToken resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
+  if (!videoToken) {
+      reject(kErrorCode, kErrorMessageDelete, nil);
+      return;
+  }
+  [BrightcovePlayerOfflineVideoManager sharedManager].delegate = self;
+  [[BrightcovePlayerOfflineVideoManager sharedManager] pauseVideoDownload:videoToken];
+  [self sendOfflineNotification];
+  resolve(nil);
+}
```
Comment (backer) [inline ios/BrightcovePlayerUtil.m:123]: nit: add spaces in between these method blocks

Comment (PratheeshDBennet) [reply ios/BrightcovePlayerUtil.m:123]: Thanks a lot for the review, Brian. Sure, will address the comments

**Comment by PratheeshDBennet** [comment]
> just one nit about spacing in between method blocks but otherwise looks good! Thanks for making these changes. I'm interested to know if there are any issues with how Android queues 1 download at a time while iOS allows for multiple, but that's something we can test once Palani's piece is done. Hopefully it is already accounted for with the base implementation.

Hi Brian, 
Understood the limitation. Yes, as far as I tested in iOS, it is supporting concurrent downloads with the legacy base implementation. However, there is also a seperate limitation section on the Brightcove iOS SDK docs which I can share across in a sheet. There is also an open bug which I will share with the BC ticket details as well in the sheet.

**Review by backer — APPROVED** [review]
👍

**Review by backer — APPROVED** [review]
👍

**Pull request closed without merging**